client/post-editor/editor-featured-image/preview-container.jsx

```diff
@@ -23,64 +22,35 @@ class EditorFeaturedImagePreviewContainer extends React.Component {
 		showEditIcon: PropTypes.bool,
 	};
 
-	state = {
-		image: null,
-	};
-
 	componentDidMount() {
-		this.fetchImage();
-		MediaStore.on( 'change', this.updateImageState );
+		this.props.fetchMediaItem( this.props.siteId, this.props.itemId );
 	}
 
 	componentDidUpdate( prevProps ) {
 		const { siteId, itemId } = this.props;
 		if ( siteId !== prevProps.siteId || itemId !== prevProps.itemId ) {
-			this.fetchImage();
+			if ( ! this.props.image ) {
+				this.props.fetchMediaItem( siteId, itemId );
+			}
+			this.props.onImageChange( itemId );
```

Comment:
Just want to note that we're removing a null check here that used to exist: https://github.com/Automattic/wp-calypso/pull/44166/files#diff-40794061a91425a8700a94c9d521ffedL69

Maybe you already checked this but I did as well, it seems like this is safe, that we always pass this prop, but just in case something mysteriously breaks on this change I wanted to point it out.

Reply:
Good catch 👍 I re-added it to be sure we don't introduce any regressions we can't foresee at this point.